File: Data Structures & Algorithms/anagram-groups/submission-4.js
class Solution {
    /**
     * @param {string[]} strs
     * @return {string[][]}
     */
    groupAnagrams(strs) {
        let pair = {}

        for (const str of strs) {
            let bucket = new Array(26).fill(0);
            for (const c of str) {
                bucket[c.charCodeAt(0) - 'a'.charCodeAt(0)] ++
            }

            let key = bucket.join('#')

            if (!(pair[key])) {
                pair[key] = []
            }

            pair[key].push(str)
        }

        return Object.values(pair)
}
}

File: Data Structures & Algorithms/longest-common-prefix/submission-4.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        res = strs[0] 

        for i in range(1, len(strs)):
            l = 0 
            r = 0 

            while l < len(res) and r < len(strs[i]): 
                if res[l] == strs[i][r]: 
                    l += 1 
                    r += 1 
                else: 
                    break 

            res = res[0 : l] 

        return res